crates/bevy_asset/src/server/info.rs

```diff
@@ -506,8 +514,10 @@ impl AssetInfos {
             if let Some(info) = self.get_mut(id) {
                 info.loading_dependencies.remove(&loaded_asset_id);
                 if info.loading_dependencies.is_empty() {
-                    // send dependencies loaded event
-                    info.dep_load_state = DependencyLoadState::Loaded;
+                    if !matches!(info.dep_load_state, DependencyLoadState::Failed(_)) {
+                        // send dependencies loaded event
+                        info.dep_load_state = DependencyLoadState::Loaded;
```

Comment:
prior to this change, the added test `dependency_load_states` would fail due to a successful dependency load overwriting a previous failed dependency load.